geode-core/src/main/java/org/apache/geode/internal/cache/DistributedCacheOperation.java

```diff
@@ -1107,6 +1107,10 @@ protected void process(final ClusterDistributionManager dm) {
 
         final LocalRegion lclRgn = getLocalRegionForProcessing(dm);
         sendReply = false;
+        if (lclRgn != null && lclRgn.getScope().isDistributedNoAck()) {
+          dm.getExecutors().getWaitingThreadPool().execute(() -> basicProcess(dm, lclRgn));
```

Comment:
this previously caught CancelException and set this.closed=true.  That's no longer being done for distributed-no-ack regions.  Why is that okay?

Reply:
I have the same concern. When a thread runs, if it throws an exception that is not caught, then that exception will just silently kill the thread. This can make it very hard to diagnose what is happening. The current thread has a few `catch` blocks and a `finally` block (line 1115-1147). If the newly spawned thread does not have the `catch` and `finally` blocks, I would like to understand why.

Another question is why use `getWaitingThreadPool()`  and not the other pool, e.g. `getThreadPool()`?

Reply:
In basicProcess call (using waiting thread pool for distributed-no-ack regions), CancelException is also caught and handled.
catch (RegionDestroyedException ignore) {
        this.closed = true;
        if (logger.isDebugEnabled()) {
          logger.debug("{} Region destroyed: nothing to do", this);
        }
      } catch (CancelException ignore) {
        this.closed = true;
        if (logger.isDebugEnabled()) {
          logger.debug("{} Cancelled: nothing to do", this);
        }
      }

Reply:
Good point. For `basicProcess()`, if debug is not enabled, some exceptions are caught, but not logged, although some flags are set, when the exceptions are caught. Will the new thread fail silently with such exceptions, if debug is not enabled?
And even for `process()`, logging the exceptions also depends on whether debug is enabled. I am not sure whether this is by design. If debug is not enabled, without the log message, it is hard to analyze the failures.

Reply:
I believe it is intentional. We do not want to pollute the user logging.

Reply:
But for this corner case, I think an info level log is necessary.